lib/Installation/ModuleSelection/v2/ModuleSelectionPage.pm

```diff
@@ -0,0 +1,68 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: The module provides interface to act with Module Selection page
+#          v2 works with YMultiSelectionBox
+#
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::ModuleSelection::v2::ModuleSelectionPage;
+use parent 'Installation::Navigation::NavigationBase';
+use strict;
+use warnings;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {
+        app => $args->{app}
+    }, $class;
+    return $self->init();
+}
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
+    $self->{sbox_addons} = $self->{app}->selectionbox({id => 'addon_repos'});
+    $self->{lbl_wizard} = $self->{app}->label({id => 'wizard'});
+    return $self;
+}
+
+sub is_shown {
+    my ($self) = @_;
+    my $result;
+    eval {
+        $result = YuiRestClient::Wait::wait_until(object => sub {
```

Comment:
I would suggest doing the wait_until (page is shown) in a different function, and keeping `is_shown` simple, but this is just my take on this